doc/services/pm/device.rst

```diff
@@ -202,58 +238,126 @@ power management implementation.
 
 Use :c:macro:`PM_DEVICE_DEFINE` or :c:macro:`PM_DEVICE_DT_DEFINE` to define the
 power management resources required by a driver. These macros allocate the
-driver-specific state which is required by the power management subsystem.
+driver-specific context which is required by the power management subsystem.
 
 Drivers can use :c:macro:`PM_DEVICE_GET` or
-:c:macro:`PM_DEVICE_DT_GET` to get a pointer to this state. These
+:c:macro:`PM_DEVICE_DT_GET` to get a pointer to this context. These
 pointers should be passed to ``DEVICE_DEFINE`` or ``DEVICE_DT_DEFINE``
 to initialize the power management field in each :c:struct:`device`.
 
-Here is some example code showing how to implement device power management
-support in a device driver.
+The following example code shows how to implement device power management
```

Comment:
I think it would be extremely useful for this to actually include some example resources that are commonly controlled in drivers (probably without error handling for brevity). As it is, the example shows the form of the callback, but doesn't really show what an implementation actually does.

I would propose two hypothetical GPIOs:
 *  One that is configured to be used as an interrupt in `ACTIVE`
 * One that controls the low power state of the device (Device is in a low power sleep mode when `GPIO_OUTPUT_ACTIVE`).

Reply:
I extended the driver example to be, detailed, maybe a bit overkill even.

Reply:
That is great, better than I hoped for.